crates/bevy_pbr/src/render/light.rs

```diff
@@ -2115,31 +2147,97 @@ impl CachedRenderPipelinePhaseItem for Shadow {
     }
 }
 
+/// The rendering node that renders meshes that were "visible" (so to speak)
+/// from a light last frame.
+///
+/// If occlusion culling for a light is disabled, then this node simply renders
+/// all meshes in range of the light.
+#[derive(Deref, DerefMut)]
+pub struct EarlyShadowPassNode(ShadowPassNode);
+
+/// The rendering node that renders meshes that became newly "visible" (so to
+/// speak) from a light this frame.
+///
+/// If occlusion culling for a light is disabled, then this node does nothing.
+#[derive(Deref, DerefMut)]
+pub struct LateShadowPassNode(ShadowPassNode);
+
+/// Encapsulates rendering logic shared between the early and late shadow pass
+/// nodes.
 pub struct ShadowPassNode {
+    /// The query that finds cameras in which shadows are visible.
     main_view_query: QueryState<Read<ViewLightEntities>>,
-    view_light_query: QueryState<(Read<ShadowView>, Read<ExtractedView>)>,
+    /// The query that finds shadow cascades.
```

Comment:
Is it necessarily only for directional light cascade shadow maps or does it work for point and spot like shadow maps too?

Reply:
It'd require more work to make this work for point lights and spot lights because they're cubemaps, so I opted to only go for directional lights for now. We can do point lights/spot lights in a followup.

Reply:
I'm actually not 100% sure that it would even be worth doing for cubemaps, because Hi-Z construction for cubemaps is going to be fairly expensive.